sources/api/schnauzer/src/helpers.rs

```diff
@@ -869,6 +881,81 @@ pub fn kube_reserve_cpu(
     Ok(())
 }
 
+/// Attempts to resolve the current hostname in DNS.  If unsuccessful, returns an entry formatted
+/// for `/etc/hosts` that aliases the hostname to both the IPV4/6 localhost addresses.
+pub fn add_unresolvable_hostname(
+    helper: &Helper<'_, '_>,
+    _: &Handlebars,
+    _: &Context,
+    renderctx: &mut RenderContext<'_, '_>,
+    out: &mut dyn Output,
+) -> Result<(), RenderError> {
+    // To give context to our errors, get the template name, if available.
+    trace!("Starting base64_decode helper");
+    let template_name = template_name(renderctx);
+    trace!("Template name: {}", &template_name);
+
+    // Check number of parameters, must be exactly one
+    trace!("Number of params: {}", helper.params().len());
+    check_param_count(helper, template_name, 1)?;
+
+    // Get the resolved key out of the template (param(0)). value() returns
+    // a serde_json::Value
+    let hostname_value = helper
+        .param(0)
+        .map(|v| v.value())
+        .context(error::Internal {
+            msg: "Found no params after confirming there is one param",
+        })?;
+    trace!("Hostname value from template: {}", hostname_value);
+
+    // Create an &str from the serde_json::Value
+    let hostname_str = hostname_value
+        .as_str()
+        .context(error::InvalidTemplateValue {
+            expected: "string",
+            value: hostname_value.to_owned(),
+            template: template_name.to_owned(),
+        })?;
+    trace!("Hostname string from template: {}", hostname_str);
+
+    // Attempt to resolve the hostname
+    let ipv4_localhost =
+        IpAddr::from_str(IPV4_LOCALHOST).context(error::IpFromString { ip: IPV4_LOCALHOST })?;
+    let ipv6_localhost =
+        IpAddr::from_str(IPV6_LOCALHOST).context(error::IpFromString { ip: IPV6_LOCALHOST })?;
+    let hostname_resolveable = match lookup_host(hostname_str) {
+        Ok(ip_list) => {
+            // If the list of IPs is empty or resolves to localhost, consider the hostname
+            // unresolvable
+            let resolves_to_localhost = ip_list
+                .iter()
+                .any(|ip| ip == &ipv4_localhost || ip == &ipv6_localhost);
+            if ip_list.is_empty() || resolves_to_localhost {
+                false
+            } else {
+                true
+            }
+        }
+        Err(e) => {
+            trace!("DNS hostname lookup failed: {},", e);
+            false
+        }
+    };
+
+    // Only write an entry to the template if the hostname is unresolvable
+    if !hostname_resolveable {
+        let ipv4_entry = format!("{} {}", IPV4_LOCALHOST, hostname_str);
+        let ipv6_entry = format!("{} {}", IPV6_LOCALHOST, hostname_str);
+        let entries = format!("{}\n{}", ipv4_entry, ipv6_entry);
```

Comment:
It might be better to ensure that a newline is written at the end of the file, rather than relying on handlebars - unless this results in an extra newline.

```suggestion
        let entries = format!("{}\n{}\n", ipv4_entry, ipv6_entry);
```

Reply:
I'd prefer not to add an extra newline since handlebars seems to add one:
```
bash-5.0# cat -E /etc/hosts 
127.0.0.1 localhost localhost.localdomain localhost4 localhost4.localdomain4$
::1 localhost localhost.localdomain localhost6 localhost6.localdomain6$
127.0.0.1 mrow-9-1$
::1 mrow-9-1$
```

Reply:
Adding the extra newline in the template adds an additional line in the rendered template:
```
bash-5.0# cat -E /etc/hosts 
127.0.0.1 localhost localhost.localdomain localhost4 localhost4.localdomain4$
::1 localhost localhost.localdomain localhost6 localhost6.localdomain6$
127.0.0.1 mrow-9-1-newline$
::1 mrow-9-1-newline$
$
```